src/fundus/scraping/article.py

```diff
@@ -16,37 +14,76 @@
-
-    @classmethod
-    def from_extracted(cls, html: HTML, extracted: Dict[str, Any], exception: Optional[Exception] = None) -> "Article":
-        validated_attributes: Set[str] = {article_field.name for article_field in fields(cls)}
-
-        extracted_unvalidated: Iterator[Tuple[str, Any]]
-        extracted_validated: Iterator[Tuple[str, Any]]
-        extracted_unvalidated, extracted_validated = more_itertools.partition(
-            lambda attribute_and_value: attribute_and_value[0] in validated_attributes, extracted.items()
-        )
+    __extraction__: Mapping[str, Any] = {}
+
+    def __init__(self, *, html: HTML, exception: Optional[Exception] = None, **extraction: Any) -> None:
+        self.html = html
+        self.exception = exception
+        self.__extraction__ = extraction
 
-        article: Article = cls(html, exception, **dict(extracted_validated))
-        unvalidated_attributes: Set[str] = set()
-        for attribute, value in extracted_unvalidated:
-            object.__setattr__(article, attribute, value)  # Sets attributes on a frozen dataclass
-            unvalidated_attributes.add(attribute)
-        object.__setattr__(article, "_unvalidated_attributes", unvalidated_attributes)
+        # create descriptors for attributes that aren't pre-defined as properties.
+        for attribute in extraction.keys():
+            if not hasattr(self, attribute):
+                setattr(self, attribute, AttributeView(attribute, self.__extraction__))
+
+    @property
+    def title(self) -> Optional[str]:
+        return self.__extraction__.get("title")
 
-        return article
+    @property
+    def body(self) -> Optional[ArticleBody]:
+        return self.__extraction__.get("body")
+
+    @property
+    def authors(self) -> List[str]:
+        return self.__extraction__.get("authors") or []
+
+    @property
+    def publishing_date(self) -> Optional[datetime]:
+        return self.__extraction__.get("publishing_date")
+
+    @property
+    def topics(self) -> List[str]:
+        return self.__extraction__.get("topics") or []
+
+    @property
+    def free_access(self) -> bool:
+        return self.__extraction__.get("free_access") or False
+
+    @property
+    def publishers(self) -> str:
+        return self.html.source_info.publisher
+
+    def __getattribute__(self, item: str):
+        if (attribute := object.__getattribute__(self, item)) and hasattr(attribute, "__get__"):
+            return attribute.__get__(self, type(self))
+        return attribute
+
+    def __setattr__(self, key: str, value: object):
+        if hasattr(self, key):
+            # we can't use getattr here, because it would invoke __get__, so unfortunately no default value
+            attribute = object.__getattribute__(self, key)
+            if hasattr(attribute, "__set__"):
+                attribute.__set__(key, value)
+                return
+        object.__setattr__(self, key, value)
+
+    def __getattr__(self, item: str):
```

Comment:
The purpose of `__getattr__` here is to prevent IDEs like pycharm from flagging attribute access for attributes not described in the constructor. In our case, every access on an unvalidated attribute would get flagged by IDEs. I did this in order to underline our intended design.